src/Console/Command/SelfUpdateCommand.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Pickle\Console\Command;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Humbug\SelfUpdate\Updater;
+
+class SelfUpdateCommand extends Command
+{
+    const PHAR_NAME = 'pickle.phar';
+
+    protected function configure()
+    {
+        $this
+            ->setName('self-update')
+            ->setDescription('Validate a PECL extension')
+            ->addOption(
+                'unstable',
+                null,
+                InputOption::VALUE_NONE,
+                'Update to a non-stable version'
+            )
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if (\Phar::running() === '') {
+            throw new \RuntimeException($this->getName() . ' can only be run from the Phar!');
+        }
+
+        $updater = new Updater(null, false, Updater::STRATEGY_GITHUB);
+        $updater->getStrategy()->setPackageName('friendsofphp/pickle');
+        $updater->getStrategy()->setPharName(self::PHAR_NAME);
+        $updater->getStrategy()->setCurrentLocalVersion('v0.5.0');
```

Comment:
well, storing the phar in github releases requires doing a github release, which means tagging a release. this does not go the rolling-release way.
Composer stores phars on their own server (btw, they support updating to releases too).

anyway, this string is not the version of the latest release. It is the version of the current code (so it should match the tag for which it gets built)